In [11]:
import pandas as pd
def wraptable(thing):
    for substr in [ 'MUTAG &   ','ER (', 'HK (', 'WC (', 'BA (']:
        thing = thing.replace(substr, f'\midrule \n{substr}', 1)
    replace = {
            'gen\\_n\\_max': "$N_\\text{max}$",
            'gen\\_n': "$N_\\text{min}$",
            'IMDB-BINARY':'IMDB-BIN',} 
    for k, v in replace.items():
        thing = thing.replace(k, v)
    return thing
    #return "\\begin{table} \n" + thing + "\n\\caption{whatever} \n" + "\\end{table}\n"


def do_bolding(latex_string, list_of_things):
    dict_of_things = {str(x): f'\\textbf{{{str(x)}}}' for x in list_of_things}
    for k, v in dict_of_things.items():
        latex_string = latex_string.replace(k, v)
    return latex_string
    

In [26]:
df = pd.read_csv('Generalizability.csv')
df.dataset = df.dataset.map(lambda x: dict(zip(['BarabasiAlbert', 'ErdosRenyi', 'PowerlawCluster', 'WattsStrogatz'], ['BA', 'ER', 'HK', 'WC'])).get(x, x))
df = df.iloc[:-3, 2:].round(2)

In [27]:
df.gen_n = df.gen_n.fillna("").apply(lambda x: int(x) if x != "" else x)
df.gen_n_max = df.gen_n_max.fillna("").apply(lambda x: int(x) if x != "" else x)

df.dataset = df.apply(lambda row: f'{row.dataset} ({row.gen_n},{row.gen_n_max})', axis=1)
df.dataset = df.dataset.apply(lambda dataset: dataset[:-4] if dataset.endswith("(,)") else dataset)

df = df[['dataset','MUTAG', 'ENZYMES', 'PROTEINS',
       'IMDB-BINARY', 'COLLAB']]

# get things to bold
list_of_things = []
for col in df.keys():
    #print(col)
    if col in ['dataset']: continue
    possibilities = [float(x.split(" ")[0]) if isinstance(x, str) else x for x in df[col].values[2:]]
    list_of_things.append(f'{min(possibilities):0.2f}')

df.rename({'dataset': 'Train Dataset'}, inplace=True, axis=1)
main_body = df.to_latex(index=False)
print(wraptable(do_bolding(main_body, list_of_things)))

\begin{tabular}{lrrrrr}
\toprule
Train Dataset &  MUTAG &  ENZYMES &  PROTEINS &  IMDB-BIN &  COLLAB \\
\midrule
  \midrule 
BA (50,100) &   \textbf{7.74} &    20.12 &     27.66 &        17.57 &   74.15 \\
 BA (100,200) &   \textbf{7.74} &    20.35 &     26.03 &        16.86 &   69.29 \\
 BA (400,500) &   8.05 &    21.00 &     26.54 &        17.34 &   70.17 \\
  \midrule 
ER (50,100) &   \textbf{7.74} &    20.37 &     28.17 &        16.86 &   69.07 \\
 ER (100,200) &   8.05 &    21.52 &     27.72 &        16.89 &   68.83 \\
 ER (400,500) &   7.79 &    21.55 &     28.60 &        16.78 &   68.74 \\
  \midrule 
HK (50,100) &   \textbf{7.74} &    20.42 &     25.60 &        17.05 &   69.17 \\
 HK (100,200) &   7.84 &    20.43 &     27.30 &        17.01 &   70.20 \\
 HK (400,500) &   7.95 &    20.63 &     26.30 &        17.15 &   69.91 \\
  \midrule 
WC (50,100) &   7.89 &    \textbf{20.13} &     25.46 &        17.38 &   70.14 \\
 WC (100,200) &   7.79 &    20.30 &     25.45 &        17.91 &

/tmp/ipykernel_1779527/3812583691.py:19: FutureWarning: In future versions `DataFrame.to_latex` is expected to utilise the base implementation of `Styler.to_latex` for formatting and rendering. The arguments signature may therefore change. It is recommended instead to use `DataFrame.style.to_latex` which also contains additional functionality.
  main_body = df.to_latex(index=False)


In [14]:
df = pd.read_csv('Generalizability.csv')
df = df.iloc[-3:, [1,3,4,5,6,7,8]].round(2).fillna("")


In [15]:
gurobi_lines = df.to_latex(index=False)
print(wraptable(gurobi_lines))

\begin{tabular}{lllrrrr}
\toprule
Unnamed: 0 & $N_\text{min}$ & $N_\text{max}$ &  MUTAG &  ENZYMES &  PROTEINS &  IMDB-BIN \\
\midrule
gurobi\_1.0 &       &           &   7.74 &     20.0 &     24.96 &        16.76 \\
gurobi\_2.0 &       &           &   7.74 &     20.0 &     24.96 &        16.76 \\
gurobi\_4.0 &       &           &   7.74 &     20.0 &     24.96 &        16.76 \\
\bottomrule
\end{tabular}



/tmp/ipykernel_1529140/2406804855.py:1: FutureWarning: In future versions `DataFrame.to_latex` is expected to utilise the base implementation of `Styler.to_latex` for formatting and rendering. The arguments signature may therefore change. It is recommended instead to use `DataFrame.style.to_latex` which also contains additional functionality.
  gurobi_lines = df.to_latex(index=False)


# instructions

1. paste the gurobi results on the bottom
2. add "\textbf{Gurobi Results} & & & & & \\" between the two segments
3. add \midrule between the segments of the main_body
4. delete gen_n for the non-generated datasets

\begin{table} 
\begin{tabular}{lrrrrrr}
\toprule
        dataset &  gen\_n &  ENZYMES &  MUTAG &  COLLAB &  PROTEINS &  IMDB-BINARY \\
\midrule
        ENZYMES &   &   -20.60 &  -7.74 &  -68.40 &    -28.29 &       -16.79 \\
          MUTAG &   &   -20.83 &  -7.74 &  -70.09 &    -26.76 &       -16.92 \\
         COLLAB &   &   -20.35 &  -7.89 &  -67.52 &    -26.13 &       -16.76 \\
       PROTEINS &   &   -20.22 &  -7.89 &  -70.26 &    -25.29 &       -16.77 \\
    IMDB-BINARY &   &   -20.97 &  -7.95 &  -68.03 &    -27.06 &       -16.76 \\
\midrule
 BarabasiAlbert &   50.0 &   -20.12 &  -7.74 &  -74.15 &    -27.66 &       -17.57 \\
 BarabasiAlbert &  100.0 &   -20.35 &  -7.74 &  -69.29 &    -26.03 &       -16.86 \\
 BarabasiAlbert &  400.0 &   -21.00 &  -8.05 &  -70.17 &    -26.54 &       -17.34 \\
\midrule
     ErdosRenyi &   50.0 &   -20.37 &  -7.74 &  -69.07 &    -28.17 &       -16.86 \\
     ErdosRenyi &  100.0 &   -21.52 &  -8.05 &  -68.83 &    -27.72 &       -16.89 \\
     ErdosRenyi &  400.0 &   -21.55 &  -7.79 &  -68.74 &    -28.60 &       -16.78 \\
\midrule
PowerlawCluster &   50.0 &   -20.42 &  -7.74 &  -69.17 &    -25.60 &       -17.05 \\
PowerlawCluster &  100.0 &   -20.43 &  -7.84 &  -70.20 &    -27.30 &       -17.01 \\
PowerlawCluster &  400.0 &   -20.63 &  -7.95 &  -69.91 &    -26.30 &       -17.15 \\
\midrule
  WattsStrogatz &   50.0 &   -20.13 &  -7.89 &  -70.14 &    -25.46 &       -17.38 \\
  WattsStrogatz &  100.0 &   -20.30 &  -7.79 &  -71.16 &    -25.45 &       -17.91 \\
  WattsStrogatz &  400.0 &   -20.48 &  -8.05 &  -70.16 &    -25.79 &       -17.12 \\
\bottomrule
\textbf{Gurobi Results} & & & & & \\
\midrule
gurobi\_1.0 &       &    -20.0 &  -7.74 &  -67.46 &    -24.96 &       -16.76 \\
gurobi\_2.0 &       &    -20.0 &  -7.74 &  -67.46 &    -24.96 &       -16.76 \\
gurobi\_4.0 &       &    -20.0 &  -7.74 &  -67.46 &    -24.96 &       -16.76 \\
\bottomrule

\end{tabular}

\caption{whatever} 
\end{table}